clients/validator/src/caches/gateways.ts

```diff
@@ -41,4 +42,18 @@ export default class GatewaysCache {
         this.gateways = newGateways
         return newGateways;
     }
+
+    /// Makes requests to assemble a full list of gateways from validator-api
+    async refreshValidatorAPIGateways(urls: string[]): Promise<GatewayBond[]> {
+        for (const url of urls) {
+            const validator_api_url = new URL(url);
+            validator_api_url.port = VALIDATOR_API_PORT;
```

Comment:
I'm not entirely sure what to feel about ignoring user-provided port and using the hardcoded one, but if somebody else takes a look at it and is fine with that, I'll also feel good about it

Reply:
The user-provided port is not ignored, but it just isn't usable for the validator-api. In the future, my thoughts are that the user shouldn't have to provide even the validator port or the `http` scheme. An IP or DNS address should be everything that's needed.